tensorflow/core/kernels/stage_op.cc

```diff
@@ -30,26 +30,82 @@ namespace {
 
 class Buffer : public ResourceBase {
  public:
-  explicit Buffer() {}
+    typedef std::vector<Tensor> Tuple;
 
-  typedef std::vector<Tensor> Tuple;
+ public:
+  explicit Buffer(int capacity) {
+    capacity_ = capacity;
+  }
+
+  bool HasBoundedCapacity() {
+    return capacity_ > 0;
+  }
+
+  bool IsFull() {
+    return buf_.size() >= capacity_;
```

Comment:
Original intention was a private helper method. Made it private.

https://github.com/tensorflow/tensorflow/pull/9686/commits/5ad80b2d5013cb521dd5fc176852748d8c6f8e89